### PR TITLE
Avoid updating the block gap when it's unchanged

### PR DESCRIPTION
**Pull request by liuchengxu**
There are basically three commits in this PR. Since all these commits essentially have no logical changes, I packed them into one PR. Review by per-commit is recommended.
- The first commit avoids unnecessarily updating the block gap storage when the value remains unchanged, as discovered when I worked on https://github.com/paritytech/polkadot-sdk/issues/5406.
- The second commit is purely about format string style changes but deletes ~10 lines of code, which slightly helps me look into this file :P
- The third commit is added to avoid the unnecessary block gap update in `BlockchainDb`.

### CODE REVIEW

File: substrate/client/db/src/lib.rs [hunk 1]
```diff
@@ -1693,19 +1693,19 @@ impl<Block: BlockT> Backend<Block> {
...
-						debug!(target: "db", "Removed block gap.");
-					} else {
-						block_gap = Some((start, end));
-						debug!(target: "db", "Update block gap. {:?}", block_gap);
-						transaction.set(
-							columns::META,
-							meta_keys::BLOCK_GAP,
-							&(start, end).encode(),
-						);
+						if start > end {
```
Comment (dmitry-markin) [inline substrate/client/db/src/lib.rs:1683]: Are you sure the block gap can't be updated externally and returned from `self.blockchain.meta.read()`?

**Review by dmitry-markin — COMMENTED** [review]
I'm not too familiar with this part of the codebase. Overall looks good, but I don't know if block gap can be updated in the db from other places.

@davxy can you have a look, please?

File: substrate/client/db/src/lib.rs (hunk 1 above)
Comment (liuchengxu) [reply substrate/client/db/src/lib.rs:1683]: Yes, I'm pretty sure that `update_block_gap()` is the only way to update the gap in meta and it's only invoked at the end of `try_commit_operation()`.

**Review by dmitry-markin — APPROVED** [review]

File: substrate/client/db/src/lib.rs [hunk 2]
```diff
@@ -1693,27 +1680,29 @@ impl<Block: BlockT> Backend<Block> {
...
+						} else {
+							block_gap = Some((start, end));
+							debug!(target: "db", "Update block gap. {block_gap:?}");
+							transaction.set(
+								columns::META,
+								meta_keys::BLOCK_GAP,
+								&(start, end).encode(),
+							);
+						}
+						block_gap_updated = true;
```
Comment (davxy) [inline substrate/client/db/src/lib.rs:1696]: Not related nit: In below condition `number > best_num + One::one()` implies `number >One::one()`.
So the second condition can be removed

**Review by davxy — APPROVED** [review]
Looks good

File: substrate/client/db/src/lib.rs (hunk 2 above)
Comment (davxy) [reply substrate/client/db/src/lib.rs:1696]: Already addressed by https://github.com/paritytech/polkadot-sdk/pull/5343, feel free to ignore

**Pull request merged**